pkg/innerring/processors/governance/list.go

```diff
@@ -0,0 +1,64 @@
+package governance
+
+import (
+	"sort"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/pkg/errors"
+)
+
+var errNotEnoughKeys = errors.New("alphabet list in mainnet is too short")
+
+// newAlphabetList returns updated list of sidechain keys with no more than 1\3
+// of new keys from mainnet list. Function returns `errNotEnoughKeys` if
+// mainnet list contains less keys than sidechain list. Function returns
+// (nil, nil) if mainnet list contains all keys from sidechain list.
+func newAlphabetList(sidechain, mainnet keys.PublicKeys) (keys.PublicKeys, error) {
+	ln := len(sidechain)
+	if len(mainnet) < ln {
+		return nil, errors.Wrapf(errNotEnoughKeys, "expecting %d keys", ln)
+	}
+
+	hmap := make(map[string]bool)
+	result := make(keys.PublicKeys, 0, ln)
+
+	for _, node := range sidechain {
+		hmap[node.Address()] = false
+	}
+
+	newNodes := 0
+	newNodeLimit := ln - (ln*2/3 + 1)
```

Comment:
Can we
```
newNodeLimit := ln/3 - 1
```
?

Reply:
We can `newNodeLimit := (ln-1)/3`, thx.

Reply:
And pls double-check potential negative `newNodeLimit` values (`ln < 3`) in the function algorithm (idk can this rly happen or not)

Reply:
I've checked that, and it should be okay. The only tricky part is with empty side chain list. It should not ever happen but I made length check in https://github.com/nspcc-dev/neofs-node/pull/421/commits/7153ed3186c46638b2b7a2bd22f3fbf35fb06f7a